process/smartContract/process.go

```diff
@@ -455,7 +455,12 @@ func (sc *scProcessor) updateDeveloperRewards(
 
 	moveBalanceGasLimit := sc.economicsFee.ComputeGasLimit(tx)
 	subMoveBalanceGasLimit := !sc.isSelfShard(tx.GetSndAddr()) && !isSmartContractResult(tx)
-	if sc.flagDeploy.IsSet() || subMoveBalanceGasLimit {
+	if !sc.flagDeploy.IsSet() && !sc.isSelfShard(tx.GetSndAddr()) {
+		usedGasByMainSC, err = safeSubUint64(usedGasByMainSC, moveBalanceGasLimit)
+		if err != nil {
+			return err
+		}
+	} else if subMoveBalanceGasLimit {
```

Comment:
What happens when subMoveBalanceGasLimit is false but sc.flagDeploy.IsSet() is true, we don't need to make the below subtraction ?

Reply:
overcomplicated this a little bit. the idea is that after update - the smart contract will receive 30% of the gas used for processing thus this sub is mandatory - it the call was not from a smart contract result. for backward compatibility - the first part should remain. I changed the else if.